server/routes/createGoal/createGoalController.ts

```diff
@@ -37,17 +42,15 @@ export default class CreateGoalController {
     } as PrisonerSummary
     req.session.prisonerSummary = prisonerSummary
 
-    const createGoalForm = req.session.createGoalForm || { prisonNumber }
-
-    const view = new CreateGoalView(prisonerSummary, createGoalForm, req.flash('errors'))
+    const view = new CreateGoalView(prisonerSummary, req.session.createGoalForm, req.flash('errors'))
     res.render('pages/goal/create/index', { ...view.renderArgs })
   }
 
   submitCreateGoalForm: RequestHandler = async (req, res, next): Promise<void> => {
     const { prisonNumber } = req.params
 
     const reviewDate = parseDate(req, 'reviewDate')
-    req.session.createGoalForm = { ...req.body, prisonNumber, reviewDate }
+    req.session.createGoalForm = { ...req.body, reviewDate }
```

Comment:
Now when the CreateGoalForm is submitted we can simply replace the session object, and it will correctly populate the prison number (because it was submitted in the HTML form submission via the hidden field)